prototype_source/inductor_cpp_wrapper_tutorial.rst

```diff
@@ -0,0 +1,88 @@
+Inductor cpp wrapper tutorial
+==============================================================
+
+**Author**: `Chunyuan Wu <https://github.com/chunyuan-w>`_, `Bin Bao <https://github.com/desertfire>`__, `Jiong Gong <https://github.com/jgong5>`__
+
+Prerequisites:
+------------
+-  `torch.compile and TorchInductor concepts in PyTorch <https://pytorch.org/tutorials/intermediate/torch_compile_tutorial.html>`__
+
+Introduction
+------------
+
+Inductor wrapper generates python code to invoke generated kernels and external kernels.
+To reduce the Python overhead of the wrapper and to serve as an intermediate step to support deployment in environments without Python, 
+we implemented cpp wrapper which generates cpp code to combine the generated and external kernels.
+
+
+API
+------------
+This feature is still in prototype stage. To turn it on, the below code change is needed:
+
+.. code:: python
+
+    import torch._inductor.config as config
+    config.cpp_wrapper = True
+
+This will speed up your models by reducing the python overhead of the inductor wrapper.
+
+
+Example code
+------------
+Taken the below frontend code as an example:
+
+.. code:: python
+    
+    import torch
+
+    def fn(x):
+        return torch.tensor(list(range(2, 40, 2)), device=x.device) + x
+
+    x = torch.randn(1)
+    opt_fn = torch.compile()(fn)
+    y = opt_fn(x)
+
+
+The main part of inductor generated code with the default python wrapper will be:
+
+.. code:: python
+
+    def call(args):
+        arg0_1, = args
+        args.clear()
+        assert_size_stride(arg0_1, (1, ), (1, ))
+        buf0 = empty_strided((19, ), (1, ), device='cpu', dtype=torch.float32)
+        cpp_fused_add_lift_fresh_0(c_void_p(constant0.data_ptr()), c_void_p(arg0_1.data_ptr()), c_void_p(buf0.data_ptr()))
+        del arg0_1
+        return (buf0, )
+
+By turning on cpp wrapper, the generated code for the ``call`` function becomes a cpp function
+``inductor_entry_cpp`` of the CPP extension ``module``:
+
+.. code:: python
```

Comment:
what about gpu triton code?

Reply:
@msaroufim I've added the gpu triton code.